### PR TITLE
Support @ALL in site lists of deploy_map

### PR DESCRIPTION
**Pull request by nvidianz**

### CODE REVIEW

File: nvflare/private/fed/server/job_meta_validator.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+        if not site_list:
+            raise ValueError(f"No site is specified in deploy_map for job {job_name}")
+
+        if ALL_SITES.casefold() in (site.casefold() for site in site_list):
+            # if @ALL is specified, no other site can be in the list
+            if len(site_list) > 1:
+                raise ValueError(f"No other site can be specified if @ALL is used for job {job_name}")
+            else:
+                site_list = [ALL_SITES]
```
Comment (chesterxgchen) [inline nvflare/private/fed/server/job_meta_validator.py:100]: else is unnecessary, as ALL_SITES is already in the site_list

File: nvflare/private/fed/server/job_meta_validator.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            if duplicates:
+                raise ValueError(f"Multiple apps to be deployed to following sites {duplicates} for job {job_name}")
+
+        return site_list
+
+    def _validate_app(self, job_name: str, meta: dict, zip_file: ZipFile) -> None:
+
+        deploy_map = meta.get(JobMetaKey.DEPLOY_MAP.value)
+
+        for app, deployment in deploy_map.items():
```
Comment (chesterxgchen) [inline nvflare/private/fed/server/job_meta_validator.py:112]: should we use deployments: plural to indicate it is not single site, but multiple sites

**Review by chesterxgchen — APPROVED** [review]
minor comments, change if you like

File: nvflare/private/fed/server/job_meta_validator.py (hunk 2 above)
Comment (nvidianz) [reply nvflare/private/fed/server/job_meta_validator.py:112]: Good idea.

File: nvflare/private/fed/server/job_meta_validator.py (hunk 1 above)
Comment (nvidianz) [reply nvflare/private/fed/server/job_meta_validator.py:100]: The purpose of this line is to use canonical form (all caps) in site_list so comparison in other code is easier.

**Review by nvkevlu — APPROVED** [review]

**Review by yhwen — APPROVED** [review]

**Pull request merged**